src/main/java/de/medizininformatikinitiative/flare/FlareApplication.java

```diff
@@ -97,4 +108,38 @@ public DiskCachingFhirQueryService diskCachingFhirQueryService(
     public Clock systemDefaultZone() {
         return Clock.systemDefaultZone();
     }
+
+    @Bean
+    @Qualifier("oauth")
+    ExchangeFilterFunction oauthExchangeFilterFunction(
+            @Value("${flare.fhir.oauth.token.uri:}") String tokenUri,
+            @Value("${flare.fhir.oauth.client.id:}") String clientId,
+            @Value("${flare.fhir.oauth.client.secret:}") String clientSecret,
+            @Value("${flare.fhir.oauth.scope:openid}") String scope) {
+        if (!tokenUri.isEmpty() && !clientId.isEmpty() && !clientSecret.isEmpty() && !scope.isEmpty()) {
+            logger.debug("Enabling OAuth2 authentication (token uri: '{}', client id: '{}', scope: '{}').",
```

Comment:
Should be info, because it's important so see how Flare is configured at startup.